packages/next/client/link.tsx

```diff
@@ -375,7 +375,7 @@ const Link = React.forwardRef<HTMLAnchorElement, LinkPropsReal>(
       onClick,
       onMouseEnter: onMouseEnterProp,
       onTouchStart: onTouchStartProp,
-      legacyBehavior = Boolean(process.env.__NEXT_NEW_LINK_BEHAVIOR) !== true,
+      legacyBehavior = false,
```

Comment:
This change doesn't seem necessary, can you provide a reproduction where it is?

Reply:
I shared the reproduction on the referenced issue: #42621

I wasn't sure if the flag should be removed completely or simply reversed as now the default behavior is not the legacy one anymore. What do you think?

Reply:
Yeah as long as the flag is kept inverting the default sounds good

Reply:
Done